0.63 changelog (#20956)
microsoft/PowerToys: README.md

Our [prioritized roadmap][roadmap] of features and utilities that the core team is focusing on.

### 0.63 - September 2022 Update

In this release, we focused on stability and improvement.

**Highlights**

- QuickAccent contains a new setting to select a language. This should reduce the number of accented characters a user needs to pick from. Thanks [@damienleroy](https://github.com/damienleroy)!

> - Reduced installer and installed sizes by sharing the Windows App SDK, VC++ redistributable and PowerToys Interop runtime files between utilities.

Reviewer: We should also reference this work will help us ultimately remove the UAC on install. Also say we will be working with getting .NET runtime work to do the same.

Reviewer: Also we could directly call out disk and file reduction here. They are pretty massive

Reviewer 2: Disk? I suppose you mean installation _file_ and/or installation?
Curious to see how much it'll be :)

Reviewer: Disk and install file is what I was referring to. The installer is much smaller and removed one dependency and our footprint on disk is much more reduced.

- The Text Extractor utility [fails to recognize text in some cases on ARM64 devices running Windows 10](https://github.com/microsoft/PowerToys/issues/20278).
- After installing PowerToys, [the new Windows 11 context menu entries for PowerRename and Image Resizer might not appear before a system restart](https://github.com/microsoft/PowerToys/issues/19124).
- There are reports of users who are [unable to open the Settings window](https://github.com/microsoft/PowerToys/issues/18015). This is being caused by incompatibilities with some applications (RTSS RivaTuner Statistics Server and MSI AfterBurner are known examples of this). If you're affected by this, please check the  linked issue to verify if any of the presented solutions works for you.

### General

- Fixed an issue that caused bug report generation to fail. (This was a hotfix for 0.62)
- Updated the Windows App SDK runtimes to 1.1.5.

### Always on Top

- Fixed an issue causing the border do linger when moving a window between virtual desktops.
- The minimum thickness for the borders is now 1. Thanks [@unuing](https://github.com/unuing)!
- Borders were showing in Virtual Desktop thumbnails. These were removed.

> - Corrected the borders visuals to more closely follow the application borders.

Reviewer 2: ```suggestion
- Corrected the borders visuals to more closely follow the application borders. Thanks [@SeraphimaZykova](https://github.com/SeraphimaZykova)!
```
This is about the _corners_, right?

Author: Yes, but Seraphima is part of the core team and not a community member. We usually call out to community members :)

### Awake

- Fixed utility exit logic to close all threads and avoid crashes. (This was a hotfix for 0.62)

### Color Picker

- Fixed utility exit logic to close all threads and avoid crashes. (This was a hotfix for 0.62)

> - Fixed initialization error that caused the mouse position to be incorrectly set.

Reviewer: Was this a community fix?

Author: It was a fix by Taras from our team.

- Fixed preview of .reg files.

### PowerToys Run

- Added support to opening Terminal windows in quake mode. Thanks [@FWest98](https://github.com/FWest98)!
- Fixed utility exit logic to close all threads and avoid crashes. (This was a hotfix for 0.62)
- Improve the icon shown in the Program plugin for application execution aliases. Thanks [@MikeBarker-MSFT](https://github.com/MikeBarker-MSFT)!
- Fix calls to the default browser when Firefox is installed from the Microsoft Store.
- Fixed accessibility issue in which controls appended to the result entries weren't announced.
- Search was improved and should now return results where the terms in the query appear at the end of the result.

### Quick Accent

- Improved the keyboard hooks performance. (This was a hotfix for 0.62)
- Fixed a bug that was causing Quick Accent to interfere with Keyboard Manager. (This was a hotfix for 0.62)
- Added the correct ß uppercase character. Thanks [@Aaron-Junker](https://github.com/Aaron-Junker)!
- Accent character selection should now wrap around. Thanks [@wmentha](https://github.com/wmentha)!

> - Added language selection setting to reduce the number of accented characters shown. Thanks [@damienleroy](https://github.com/damienleroy)!

Reviewer 3: we should mention all the new languages added.

### Screen Ruler

- Improved UI/UX and settings descriptions.
- Fixed utility exit logic to close all threads and avoid crashes. (This was a hotfix for 0.62)

### Settings

- UI icons updated. Thanks [@Jay-o-Way](https://github.com/Jay-o-Way)!
- Descriptions improvement and disambiguation. Thanks [@Jay-o-Way](https://github.com/Jay-o-Way)!
- Fixed checkbox margins and other design tweaks. Thanks [@Jay-o-Way](https://github.com/Jay-o-Way)!

### Text Extractor

- Removed extra spaces when recognizing Chinese, Japanase or Korean languages. Thanks [@TheJoeFin](https://github.com/TheJoeFin)!
- Fixed utility exit logic to close all threads and avoid crashes. (This was a hotfix for 0.62)
- Fixed an issue where a selection would start on right-click.

### Installer

- Added logic to exit PowerToys on upgrade before trying to update .NET.
- Updated the .NET dependency to 6.0.9.

> - Added clearer installation step names for the boostrapper.

Reviewer 2: @yuyoyuppe & @Jay-o-Way :)

Reviewer 3: @jay-o-way
not only you helped here.

Author: @yuyoyuppe is part of the core team :) 
I'll add @Jay-o-Way  and @htcfreek , though.

Author: I've pushed the suggested changes.

- Windows App SDK, VC++ redistributable and PowerToys Interop runtime files are now shared between utilites through hardlinks, reducing installation size.

### Documentation

- Fixed typos in Keyboard Manager documentation. Thanks [@eltociear](https://github.com/eltociear)!
- Replaced docs.microsoft.com links with learn.microsoft.com. Thanks [@Aaron-Junker](https://github.com/Aaron-Junker)!

### Development

- Fixed a build error that was restricting developers to switch between configuration without first cleaning local build files.

> - C++ exception catches were corrected to be caught by reference to avoind unneccessary copy operations. Thanks [@NN---](https://github.com/NN---)!

Reviewer 2: ```suggestion
- C++ exception catches were corrected to be caught by reference to avoid unneccessary copy operations. Thanks [@NN---](https://github.com/NN---)!
```

- General C# code clean up, format fixing and removal of unused code analysis suppressions.
- Removed unneccessary `muxc` prefix from XAML files. Thanks [@Jay-o-Way](https://github.com/Jay-o-Way)!

> - Re-enabled tests that depend on WebView2.

Reviewer 4: ```suggestion
- Re-enabled tests on our pipeline that depend on WebView2.
```

- Windows 11 tier 1 context menu packages now contain the "Microsoft.PowerToys" prefix.

#### What is being planned for v0.64

For [v0.64][github-next-release-work], we'll work on below:

- Environment Variables Editor PowerToy
- GPO policies for PowerToys

> - Utility to list which processes are using a file

Reviewer: We could call the community lead Hosts file app + backup for settings.

Author: I've pushed the suggested changes.

Our [prioritized roadmap][roadmap] of features and utilities that the core team is focusing on.

### 0.63 - September 2022 Update

In this release, we focused on stability and improvement.

**Highlights**

- QuickAccent contains a new setting to select a language. This should reduce the number of accented characters a user needs to pick from. Thanks [@damienleroy](https://github.com/damienleroy)!

> - Reduced installer (125 MB in 0.62.1 compared to 83 MB in 0.63.0) and installed (817 MB in 0.62.1 compared to 587 MB in 0.63.0) sizes by sharing the Windows App SDK, VC++ redistributable and PowerToys Interop runtime files between utilities. This is a step towards removing the UAC requirement on install. The next step is shipping .NET self-contained and shared between utilities.

Reviewer 2: Just a suggestion.
```suggestion
- Reduced installer file size (83 MB for 0.63.0 compared to 125 MB for 0.62.1) and drive storage use (587 MB for 0.63.0 compared to 817 MB for 0.62.1) sizes by sharing the Windows App SDK, VC++ redistributable and PowerToys Interop runtime files between utilities. This is a step towards removing the UAC requirement on install. The next step is shipping .NET self-contained and shared between utilities.
```

### Image Resizer

- Fixed a bug causing File Explorer to crash under some conditions when accessing the context menu.

### PowerToys Run

- Added support to opening Terminal windows in quake mode. Thanks [@FWest98](https://github.com/FWest98)!
- Fixed utility exit logic to close all threads and avoid crashes. (This was a hotfix for 0.62)
- Improve the icon shown in the Program plugin for application execution aliases. Thanks [@MikeBarker-MSFT](https://github.com/MikeBarker-MSFT)!
- Fix calls to the default browser when Firefox is installed from the Microsoft Store.
- Fixed accessibility issue in which controls appended to the result entries weren't announced.
- Search was improved and should now return results where the terms in the query appear at the end of the result.

### Quick Accent

- Improved the keyboard hooks performance. (This was a hotfix for 0.62)
- Fixed a bug that was causing Quick Accent to interfere with Keyboard Manager. (This was a hotfix for 0.62)
- Added the correct ß uppercase character. Thanks [@Aaron-Junker](https://github.com/Aaron-Junker)!
- Accent character selection should now wrap around. Thanks [@wmentha](https://github.com/wmentha)!

> - Added language selection setting to reduce the number of accented characters shown. The available languages are Currency, Czech, Dutch, French, Hungarian, Icelandic, Italian, Maori, Pinyin, Polish, Romanian, Slovakian, Spanish and Turkish. Thanks [@damienleroy](https://github.com/damienleroy)!

Reviewer 3: Germany is missing. Did you write Dutch instead of Germany?

Reviewer 2: Sharp one.
Also: [Pinyin](https://wikipedia.org/wiki/Pinyin) is not a **language**. It's a way to transliterate Chinese (possibly Taiwanese) to roman alphabet.
@damienleroy I think it would be even better to name these _subsets_ in stead of languages. Also looking at "Currency"...

Author: I see no German, there 🤷  
![image](https://user-images.githubusercontent.com/26118718/193681629-57c68ba9-c9f1-4a36-93c0-38d9299c1d3b.png)

Author: Guess it might be better to just remove the language list from the notes at this time?

Author: Or just call them character sets?

Author: Changed it to "The available accented character sets are ..."

Reviewer 2: Have a look over at Damien's repo and there we see German (DE) and no Dutch (NL). Must have gone lost in translation, so to say 😅

### Screen Ruler

- Improved UI/UX and settings descriptions.
- Fixed utility exit logic to close all threads and avoid crashes. (This was a hotfix for 0.62)

### Settings

- UI icons updated. Thanks [@Jay-o-Way](https://github.com/Jay-o-Way)!
- Descriptions improvement and disambiguation. Thanks [@Jay-o-Way](https://github.com/Jay-o-Way)!
- Fixed checkbox margins and other design tweaks. Thanks [@Jay-o-Way](https://github.com/Jay-o-Way)!

### Text Extractor

- Removed extra spaces when recognizing Chinese, Japanase or Korean languages. Thanks [@TheJoeFin](https://github.com/TheJoeFin)!
- Fixed utility exit logic to close all threads and avoid crashes. (This was a hotfix for 0.62)
- Fixed an issue where a selection would start on right-click.

### Installer

- Added logic to exit PowerToys on upgrade before trying to update .NET.
- Updated the .NET dependency to 6.0.9.
- Added clearer installation step names for the boostrapper. Thanks [@htcfreek](https://github.com/htcfreek) and [@Jay-o-Way](https://github.com/Jay-o-Way)!

> - Windows App SDK, VC++ redistributable and PowerToys Interop runtime files are now shared between utilites through hardlinks, reducing installation size.

Reviewer 5: ```suggestion
- Windows App SDK, VC++ redistributable and PowerToys Interop runtime files are now shared between utilities through hardlinks, reducing installation size.
```

### Documentation

- Fixed typos in Keyboard Manager documentation. Thanks [@eltociear](https://github.com/eltociear)!
- Replaced docs.microsoft.com links with learn.microsoft.com. Thanks [@Aaron-Junker](https://github.com/Aaron-Junker)!

### Development

- Fixed a build error that was restricting developers to switch between configuration without first cleaning local build files.

> - C++ exception catches were corrected to be caught by reference to avoid unneccessary copy operations. Thanks [@NN---](https://github.com/NN---)!

Reviewer 5: ```suggestion
- C++ exception catches were corrected to be caught by reference to avoid unnecessary copy operations. Thanks [@NN---](https://github.com/NN---)!
```

- General C# code clean up, format fixing and removal of unused code analysis suppressions.

> - Removed unneccessary `muxc` prefix from XAML files. Thanks [@Jay-o-Way](https://github.com/Jay-o-Way)!

Reviewer 5: ```suggestion
- Removed unnecessary `muxc` prefix from XAML files. Thanks [@Jay-o-Way](https://github.com/Jay-o-Way)!
```
